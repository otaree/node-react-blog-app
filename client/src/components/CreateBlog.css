.CreateBlog h2 {
    text-align: center;
}