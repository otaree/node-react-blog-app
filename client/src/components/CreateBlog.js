import React from 'react';
import { connect } from 'react-redux';
import Form from './Form';
import * as actions from '../actions/blog';
import "./CreateBlog.css";

export class CreateBlog extends React.Component {
    onSubmit = (blogData) => {
        this.props.createBlog(blogData);
        this.props.history.push('/');
    }
    render() {
        return (
            <section className="CreateBlog">
                <h2>Create Blog</h2>
                <Form onSubmit={this.onSubmit} />
            </section>
        );
    }
}


const mapDispatchToProps = dispatch => {
    return {
        createBlog: (blog) => dispatch(actions.createBlog(blog))
    }
}

export default connect(null, mapDispatchToProps)(CreateBlog);