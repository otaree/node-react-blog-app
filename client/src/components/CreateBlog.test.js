import React from 'react';
import { shallow } from 'enzyme';
import { CreateBlog } from './CreateBlog';

describe('CreateBlog', () => {
    const mockCreateBlog = jest.fn();
    const mockPush = jest.fn();
    const props = {
        createBlog: mockCreateBlog,
        history: {
            push: mockPush
        }
    };

    const createBlog = shallow(<CreateBlog {...props} />);

    it('renders correctly', () => {
        expect(createBlog).toMatchSnapshot();
    });

    it('should handle `createBlog`', () => {
        const blogData = {
            title: "Test Title",
            body: "Test body",
            author: "tester",
            published: false
        };
        createBlog.find('Form').prop('onSubmit')(blogData);
        expect(mockPush).toHaveBeenLastCalledWith('/');
        expect(mockCreateBlog).toHaveBeenLastCalledWith(blogData);
    });
});