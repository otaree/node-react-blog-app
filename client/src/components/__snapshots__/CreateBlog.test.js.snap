// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBlog renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateBlog
    createBlog={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Create Blog
</h2>,
        <Form
          onSubmit={[Function]}
/>,
      ],
      "className": "CreateBlog",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create Blog",
        },
        "ref": null,
        "rendered": "Create Blog",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Create Blog
</h2>,
          <Form
            onSubmit={[Function]}
/>,
        ],
        "className": "CreateBlog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Blog",
          },
          "ref": null,
          "rendered": "Create Blog",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
